recipe/install_nvcc.sh

```diff
@@ -65,8 +65,13 @@ fi
 
 if [[ -z "\$(\${CUDA_HOME}/bin/nvcc --version | grep "Cuda compilation tools, release ${PKG_VERSION}")" ]]
 then
-  echo "Version of installed CUDA didn't match package"
-  return 1
+  if [[ ! -z "\${CONDA_BUILD_SYSROOT+x}" ]]
```

Comment:
Sorry I just woke up and am bad at negations, but do we need `!`?

Reply:
Shouldn't it just be
```
if [[ -n "${CONDA_BUILD_SYSROOT}" ]]
```
?

Reply:
I think the problem is in `+x`. Could you do a quick check in your terminal?

Reply:
We want this if branch to be not hit if `CONDA_BUILD_SYSROOT` is not set.

Reply:
Oh it's the backslash `\`...

Reply:
While looking into this I found that `CONDA_BUILD_SYSROOT` is being set when `sysroot_linux-64` is present in the conda environment. I was under an assumption this is only set during conda-build process and not in a user conda-environment. @isuruf is this an expected behavior?

Reply:
I like `CONDA_BUILD_STATE`. We also used it in msmpi-feedstock to detect if we're using conda-build.

Reply:
Could you please clarify how you would like to use `CONDA_BUILD_STATE` here, @jaimergp?

Reply:
We want to preserve the current behavior (to fail if the version mismatches) but only inside the conda-build env (on conda-forge).

Reply:
In that case it sounds like checking `CONDA_BUILD` would be sufficient

Reply:
Updated the script to use `CONDA_BUILD`